src/runtime/contrib/tensorrt/tensorrt_runtime.cc

```diff
@@ -106,9 +104,11 @@ class TensorRTRuntime : public JSONRuntimeBase {
 #ifdef TVM_GRAPH_RUNTIME_TENSORRT
   /*! \brief Run inference using built engine. */
   void Run() override {
+    BuildEngine();
```

Comment:
Is the reason of moving `BuildEngine` from `Init` to `Run` because you need subgraph specific information (e.g., I/O data entry IDs) to allocate device buffers?

Reply:
Thanks @comaniac for the review! Yes, to allocate the device buffers we need the DLTensor context and shape. `data_entry_` in JSON runtime isn't initialized until `Run()` so I had to move BuildEngine.

In the future, we are planning to be able to dynamically build engines for different input shapes in order to handle subgraphs with dynamic input sizes, so moving it would be needed for that anyway.